### ISSUE TITLE
Five "otel_*" options appear twice in the "Configuration Reference"

### ISSUE THREAD

**Issue by bart113** [apache/airflow#43366]
### What do you see as an issue?

In https://github.com/apache/airflow/commit/ccf12026b44685f2d9917cd313676784e63c81af five configuration options have been moved from the "metrics" section to the "traces" section. Unfortunately, they have also been left in "metrics" untouched, and now they are effectively duplicated in the documentation. This is unfortunate, because users cannot easily tell what's the difference between them and which option should they prefer to use.

Here's a list of the duplicated options:

- otel_debugging_on
- otel_host
- otel_on
- otel_port
- otel_ssl_active

If you search for them on this page: https://airflow.apache.org/docs/apache-airflow/2.10.2/configurations-ref.html, you'll see that each of them occurs twice.

### Solving the problem

The problem can be solved by removing the `[metrics] otel_*` options from the `airflow/config_templates/config.yml` file (and keeping the `[traces] otel_*` options intact).

### Anything else

The newer options also have descriptions that do not render as nicely as the old ones in the documentation (the markdown highlights are gone). It may be a good moment to fix this.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Can you please fix it maybe and make a PR for that ? Better fix should be to add them to deprecated options here: https://github.com/apache/airflow/blob/main/airflow/configuration.py#L347 - otherwise I marked it as "good first issue" and it will have to wait for someone to pick it up.

**Comment by InderParmar** [comment]
Hi, I am eager to work on this issue, can you please assign this issue to me, i will start working on it asap, and can you do me a favour by adding the "hacktoberfest" label to this issue.

**Comment by InderParmar** [comment]
Hi, can you please review the PR, and if possible, can you please add the "Hacktoberfest" label on this issue, that will be great.
Also, please let me know if there are any concerns regarding the PR. 
Here is the summary of the changes done:
- **Removed Duplicated Options**: The following otel_* options were removed from the [metrics] section in config.yml:
  - otel_debugging_on
  - otel_host
  - otel_on
  - otel_port
  - otel_ssl_active
  
- **Formatting Update in Traces Section**: Improved markdown consistency in the [traces] section for better documentation clarity.

- **Deprecated Options Added**: Added otel_* options to the deprecated list in configuration.py, directing users to the correct [traces] section options.

**Comment by ferruzzi** [comment]
They weren't moved, there is a set of options for the metrics and a set for traces, they are two separate features which CAN both handled by otel but are not necessarily.  A user could use statsd for metrics and otel for traces, or use otel for metrics and not want to enable traces at all, or have metrics and traces sent to two different frontend clients, etc.    Some change may be suitable, but simply removing the metrics settings is not the right answer.

**Comment by potiuk** [comment]
@InderParmar We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.

**Comment by GirishRGowda** [comment]
Given @ferruzzi's point that metrics and traces are separate features each needing their own otel_* options, would a PR that keeps both sections but rewrites the descriptions to clearly distinguish them (and fixes the markdown rendering in [traces]) be the accepted resolution here? Happy to pick it up on that basis — also curious why #67081 didn't land, in case there's context I should know.

**Comment by ferruzzi** [comment]
Yeah, if you want to rephrase the docs for a bit more clarity I think that could be helpful.  It looks to me like 67081 was just abandoned?

**Comment by GirishRGowda** [comment]
Thanks! I'll rephrase the descriptions so the [metrics] and [traces] otel_* options clearly state which signal each configures, and fix the markdown rendering in [traces]. Will open a PR shortly.

**Comment by br413** [comment]
Picking this up following @ferruzzi's guidance (keep both \[metrics]\ and \[traces]\ \otel_*\ options; clarify which signal each configures) and the approach in closed #70837.\n\nPlan:\n- Fix incorrect cross-signal wording in \[metrics]\ (\otel_host\, \otel_interval_milliseconds\, \otel_service\)\n- Add explicit metrics-vs-traces notes on \otel_on\ in both sections\n- Restore literal markup in \[traces]\ (\\True\\, \\config.yml\\, etc.)\n\nOpening a PR shortly.

**Comment by potiuk** [comment]
Your messsages are automated on this and previous issue - and asking for assignment means that you have not read the contribution guidelines (we do not assign issues) - you are free to work on it as you wish though - but part of the contribution is working out a good way of implementing it - without necessarily AI involved in discussing it, present YOUR (not your AI) understanding of the issue and Airflow in the PR .. Start from that.
